src/Core/src/Handlers/Image/ImageHandler.Windows.cs

```diff
@@ -8,33 +8,123 @@ namespace Microsoft.Maui.Handlers
 {
 	public partial class ImageHandler : ViewHandler<IImage, Image>
 	{
+		/// <inheritdoc/>
 		protected override Image CreatePlatformView() => new Image();
 
+		/// <inheritdoc/>
 		protected override void DisconnectHandler(Image platformView)
 		{
 			base.DisconnectHandler(platformView);
 			SourceLoader.Reset();
 		}
 
+		/// <inheritdoc/>
 		public override bool NeedsContainer =>
 			VirtualView?.Background != null ||
+			VirtualView?.Aspect == Aspect.AspectFill ||
 			base.NeedsContainer;
 
+		/// <inheritdoc/>
+		protected override void SetupContainer()
+		{
+			base.SetupContainer();
+
+			// VerticalAlignment only works when the child's Height is Auto
+			PlatformView.Height = Primitives.Dimension.Unset;
+
+			UpdateValue(nameof(IView.Height));
+			UpdateValue(nameof(IView.Width));
+		}
+
+		/// <inheritdoc/>
+		protected override void RemoveContainer()
+		{
+			base.RemoveContainer();
+
+			UpdateValue(nameof(IView.Height));
+			UpdateValue(nameof(IView.Width));
+		}
+
+		/// <summary>
+		/// Maps the abstract <see cref="IView.Height"/> property to the platform-specific implementations.
+		/// </summary>
+		/// <param name="handler">The associated handler.</param>
+		/// <param name="view">The associated <see cref="Image"/> instance.</param>
+		public static void MapHeight(IImageHandler handler, IImage view)
+		{
+			// VerticalAlignment only works when the container's Height is set and the child's Height is Auto. The child's Height
+			// is set to Auto when the container is introduced.
+			if (handler.ContainerView is FrameworkElement container)
+			{
+				container.Height = view.Height;
+				handler.PlatformView.Height = Primitives.Dimension.Unset;
+			}
+			else
+			{
+				ViewHandler.MapHeight(handler, view);
```

Comment:
Once we merge [PR](https://github.com/dotnet/maui/pull/17918),  we can rebase this PR to use the new extention methods.